ssl/ssl_lib.c

```diff
@@ -972,6 +972,10 @@ SSL *ossl_ssl_connection_new_int(SSL_CTX *ctx, const SSL_METHOD *method)
     s->ext.ech.innerch1 = NULL;
     s->ext.ech.encoded_innerch = NULL;
     s->ext.ech.kepthrr = NULL;
+#endif
+#ifndef OPENSSL_NO_SECH
+    s->sech.symmetric_key = OPENSSL_malloc(SECH_SYMMETRIC_KEY_MAX_LENGTH);
+    s->sech.symmetric_key = ctx->sech.symmetric_key;
```

Comment:
This is wrong. You're discarding the allocated pointer, so it can never be freed. Need to do a deep copy here, not just copy the pointer, likely with `memcpy(dest_ptr, from_ptr, length)`